### PR TITLE
Preparation to reuse future common dpctl f/w in functions from `vm` extension

### PR DESCRIPTION
**Pull request by antonwolfy**
The PR makes implementation changes in `vm` extension to further reuse of dpctl tensor f/w around `py_unary_ufunc` and `py_binary_ufunc` functions.
Where the new added files under `dpnp/backend/extensions/elementwise_functions` folder should be replaced by proper dpctl tensor headers.
The same f/w can be later reused while adding implementation of elementwise functions which are not available in dpctl.

Also the PR splits each function to separate dedicated source and header files instead of single big source file.

- [x] Have you provided a meaningful PR description?
- [ ] Have you added a test, reproducer or referred to issue with a reproducer?
- [ ] Have you tested your changes locally for CPU and GPU devices?
- [ ] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [x] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: dpnp/backend/extensions/vm/abs.cpp [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+void init_abs(py::module_ m)
+{
+    using arrayT = dpctl::tensor::usm_ndarray;
+    using event_vecT = std::vector<sycl::event>;
+
+    impl::populate_dispatch_vectors();
+    using impl::contig_dispatch_vector;
+    using impl::output_typeid_vector;
+
+    auto abs_pyapi = [&](sycl::queue exec_q, arrayT src, arrayT dst,
```
Comment (vtavana) [inline dpnp/backend/extensions/vm/abs.cpp:114]: Should we pass by reference?
```suggestion
    auto abs_pyapi = [&](sycl::queue exec_q, const arrayT &src, const arrayT &dst,
```

File: dpnp/backend/extensions/vm/abs.cpp [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            td_ns::NullPtrVector<impl::unary_strided_impl_fn_ptr_t>{});
+    };
+    m.def("_abs", abs_pyapi,
+          "Call `abs` function from OneMKL VM library to compute "
+          "the absolute value of vector elements",
+          py::arg("sycl_queue"), py::arg("src"), py::arg("dst"),
+          py::arg("depends") = py::list());
+
+    auto abs_need_to_call_pyapi = [&](sycl::queue exec_q, arrayT src,
+                                      arrayT dst) {
```
Comment (vtavana) [inline dpnp/backend/extensions/vm/abs.cpp:129]:
```suggestion
    auto abs_need_to_call_pyapi = [&](sycl::queue &exec_q, const arrayT &src,
                                      const arrayT &dst) {
```

File: dpnp/backend/extensions/vm/common.hpp [hunk 3]
```diff
@@ -25,252 +25,46 @@
...
-    sycl::queue exec_q,
-    dpctl::tensor::usm_ndarray src1,
-    dpctl::tensor::usm_ndarray src2,
-    dpctl::tensor::usm_ndarray dst, // dst = op(src1, src2), elementwise
-    const std::vector<sycl::event> &depends,
-    const dispatchT &dispatch_vector)
+template <typename output_typesT, typename contig_dispatchT>
+bool need_to_call_unary_ufunc(sycl::queue &exec_q,
+                              dpctl::tensor::usm_ndarray &src,
+                              dpctl::tensor::usm_ndarray &dst,
```
Comment (vtavana) [inline dpnp/backend/extensions/vm/common.hpp:51]:
```suggestion
                              const dpctl::tensor::usm_ndarray &src,
                              const dpctl::tensor::usm_ndarray &dst,
```

File: dpnp/backend/extensions/vm/common.hpp [hunk 4]
```diff
@@ -338,26 +132,35 @@ bool need_to_call_unary_ufunc(sycl::queue exec_q,
...
-bool need_to_call_binary_ufunc(sycl::queue exec_q,
-                               dpctl::tensor::usm_ndarray src1,
-                               dpctl::tensor::usm_ndarray src2,
-                               dpctl::tensor::usm_ndarray dst,
-                               const dispatchT &dispatch_vector)
+template <typename output_typesT, typename contig_dispatchT>
+bool need_to_call_binary_ufunc(sycl::queue &exec_q,
+                               dpctl::tensor::usm_ndarray &src1,
+                               dpctl::tensor::usm_ndarray &src2,
+                               dpctl::tensor::usm_ndarray &dst,
```
Comment (vtavana) [inline dpnp/backend/extensions/vm/common.hpp:145]:
```suggestion
                               const dpctl::tensor::usm_ndarray &src1,
                               const dpctl::tensor::usm_ndarray &src2,
                               const dpctl::tensor::usm_ndarray &dst,
```

File: dpnp/backend/extensions/vm/add.cpp [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+{
+    using arrayT = dpctl::tensor::usm_ndarray;
+    using event_vecT = std::vector<sycl::event>;
+
+    impl::populate_dispatch_tables();
+    using impl::contig_dispatch_vector;
+    using impl::output_typeid_vector;
+
+    auto add_pyapi = [&](sycl::queue exec_q, arrayT src1, arrayT src2,
+                         arrayT dst, const event_vecT &depends = {}) {
```
Comment (vtavana) [inline dpnp/backend/extensions/vm/add.cpp:137]:
```suggestion

    auto add_pyapi = [&](sycl::queue &exec_q, const arrayT &src1, const arrayT &src2,
                         const arrayT &dst, const event_vecT &depends = {}) {
```

File: dpnp/backend/extensions/vm/add.cpp [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+    };
+    m.def("_add", add_pyapi,
+          "Call `add` function from OneMKL VM library to performs element "
+          "by element addition of vector `src1` by vector `src2` "
+          "to resulting vector `dst`",
+          py::arg("sycl_queue"), py::arg("src1"), py::arg("src2"),
+          py::arg("dst"), py::arg("depends") = py::list());
+
+    auto add_need_to_call_pyapi = [&](sycl::queue exec_q, arrayT src1,
+                                      arrayT src2, arrayT dst) {
```
Comment (vtavana) [inline dpnp/backend/extensions/vm/add.cpp:159]:
```suggestion
    auto add_need_to_call_pyapi = [&](sycl::queue &exec_q, const arrayT &src1,
                                      const arrayT &src2, const arrayT &dst) {
```

File: dpnp/backend/extensions/vm/abs.cpp (hunk 1 above)
Comment (antonwolfy) [reply dpnp/backend/extensions/vm/abs.cpp:114]: Yep, definitely, good catch!

**Review by vtavana — APPROVED** [review]
Thank you, @antonwolfy!

**Pull request merged**